### DISCUSSION TITLE
make problem on Windows

### DISCUSSION THREAD

**Discussion by Flower-Wang1024** [cvc5/cvc5#10428] [Q&A] [answered]
hello!I meet a problem when I run 'make', can you help me solve it? Thanks!!!
![1708778770936](https://github.com/cvc5/cvc5/assets/62416914/ca9a18b2-1c4d-4908-a00b-873508073017)

**Comment by aytey** [comment]
I think I’ve seen this before.

Was it to do with having libpoly enabled and that “failing” but not in a way the “here’s how to build” gets printed?

Can you share the whole output of your `./configure.sh` invocation?

**Reply by Flower-Wang1024** [reply]
Sure, this is whole output. I know to run 'make' need makefile, but I can't find it. Is it because the address is wrong?
![1708825524882](https://github.com/cvc5/cvc5/assets/62416914/ee8116d8-d160-4ac2-8215-f277c24d813d)
![image](https://github.com/cvc5/cvc5/assets/62416914/86d7f980-003e-46d9-b50b-7796e5591f44)

**Reply by aytey** [reply]
Oh, this isn’t the bug I expected.

CMake is picking up MSVC as your compiler, and is therefore creating an MSBuild project.

Can you try the following:

```
export CC=gcc
export CXX=g++
export CMAKE_GENERATOR=“MSYS Makefiles”
```

and then delete your `build` folder and try running `./configure.sh` again?

**Reply by Flower-Wang1024** [reply]
thanks!!!

**Comment by daniel-larraz** [comment, marked as answer]
The `--win64` option is employed for cross-compiling cvc5 to target Windows 64-bit from a Linux environment. However, for native compilation of cvc5 on Windows, you need to use the `--win64-native` option with `configure.sh` instead.

Native compilation of binaries on Windows is not officially supported yet (though we are working on it). But you can try it with the latest version from the main branch using [MSYS2](https://www.msys2.org/). To do so, ensure you have MSYS2 installed, along with the necessary packages for building cvc5 (`pacman -S make mingw-w64-x86_64-cmake mingw-w64-x86_64-gcc mingw-w64-x86_64-gmp`). After setting up MSYS2, launch the [MINGW64 environment](https://www.msys2.org/docs/environments/) and execute `configure.sh` with the `--win64-native` option as described above.

**Reply by Flower-Wang1024** [reply]
It seems work! Thanks!

**Reply by Flower-Wang1024** [reply]
If I want to use python-bindings, what instructions should I execute?

**Reply by daniel-larraz** [reply]
Support for Python bindings on Windows is still missing, but we are currently working on it. We should have an update in a couple of weeks.

**Reply by daniel-larraz** [reply]
Following up on this. With the latest release (1.2.0), we now provide support for Python bindings on Windows, along with official PyPI packages for that platform. Moreover, the `--win64-native` option is no longer required when running `configure.sh` in an MSYS2 environment.
